aips/aip-77.md

```diff
@@ -0,0 +1,53 @@
+---
+aip: 77
+title: Multisig V2 Enhancement
+author: junkil-park (https://github.com/junkil-park), movekevin (https://github.com/movekevin)
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Standard (Framework)
+created: 3/28/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-77 - Multisig V2 Enhancement
+
+## Summary
+
+This AIP proposes to enhance the Multisig V2 by (1) limiting the maximum number of pending transactions, (2) introducing batch operation functions, and (3) implicitly voting on the transaction execution.
+
+# High-level Overview
+
+This AIP proposes to enhance the Multisig V2 by adding the following features:
+1. Limit the maximum number of pending transactions. This is done by introducing and enforcing the new constant `MAX_PENDING_TRANSACTIONS` in the Multisig V2 account module, which is set at 20. If the number of pending transactions exceeds this limit, the contract will not accept any new transaction proposals until the number of pending transactions falls below the limit. This is to prevent the pending transaction queue from being overloaded with too many pending transactions, thus effectively mitigating the risk of a DoS attack (as described in https://github.com/aptos-labs/aptos-core/issues/8411).
```

Comment:
Yes, we can still vote and execute, even if it's already too full. However, it will fail (abort) only when we try to add a new transaction when it's too full.